paasta_tools/autoscaling/cluster_boost.py

```diff
@@ -0,0 +1,160 @@
+
+
+def is_boost_active(
+    region: str,
+    pool: str,
+    zk: ZookeeperPool,
+) -> bool:
+    current_time = int(datetime.now().strftime('%s'))
+    end_time, _, _ = get_boost_values(region, pool, zk)
+    return current_time < end_time
+
+
+def get_boost_values(
+    region: str,
+    pool: str,
+    zk: ZookeeperPool,
+) -> (int, float, float):
+    # Default values, non-boost.
+    end_time = 0
+    boost_factor = 1.0
+    expected_load = 0
+
+    try:
+        zk_boost_path = get_zk_boost_path(region, pool)
+        end_time = int(zk.get(zk_boost_path + '/end_time')[0].decode('utf-8'))
+        boost_factor = float(zk.get(zk_boost_path + '/factor')[0].decode('utf-8'))
+        expected_load = float(zk.get(zk_boost_path + '/expected_load')[0].decode('utf-8'))
+
+    except Exception as e:
+        log.info(e)
+        log.debug('No boost data in Zookeeper for pool {} in region {}.'.format(
+            pool,
+            region,
+        ))
+
+    return end_time, boost_factor, expected_load
+
+
+def set_boost_factor(
+    region: str,
+    pool: str,
+    factor=DEFAULT_BOOST_FACTOR,
+    duration_minutes=DEFAULT_BOOST_DURATION,
+    override=False,
+) -> bool:
+    if factor < MIN_BOOST_FACTOR:
+        raise ValueError('Cannot set a boost factor smaller than {}'.format(MIN_BOOST_FACTOR))
+
+    if factor > MAX_BOOST_FACTOR:
+        raise ValueError('Boost factor > {} does not sound reasonable'.format(MAX_BOOST_FACTOR))
+
+    if duration_minutes > MAX_BOOST_DURATION:
+        raise ValueError(
+            'Boost duration > {} minutes does not sound reasonable'.format(MAX_BOOST_DURATION),
+        )
+
+    zk_boost_path = get_zk_boost_path(region, pool)
+    current_time = int(datetime.now().timestamp())
+    end_time = current_time + 60 * duration_minutes
+
+    zk_end_time_path = zk_boost_path + '/end_time'
+    zk_factor_path = zk_boost_path + '/factor'
+    zk_expected_load_path = zk_boost_path + '/expected_load'
+
+    with ZookeeperPool() as zk:
+        if not override and is_boost_active(region, pool, zk):
+            raise BoostAlreadyActiveError
+
+        try:
+            zk.ensure_path(zk_end_time_path)
+            zk.ensure_path(zk_factor_path)
+            zk.ensure_path(zk_expected_load_path)
+            zk.set(zk_end_time_path, str(end_time).encode('utf-8'))
+            zk.set(zk_factor_path, str(factor).encode('utf-8'))
+            zk.set(zk_expected_load_path, '0'.encode('utf-8'))
+        except Exception:
+            log.error('Error setting the boost in Zookeeper')
+            raise
+
+        log.info('Cluster boost: Setting capacity boost factor {} for pool {} in region {} until {}'.format(
+            factor,
+            pool,
+            region,
+            datetime.fromtimestamp(end_time).strftime('%Y-%m-%d %H:%M:%S UTC'),
+        ))
+
+        # Let's check that this factor has been properly written to zk
+        return get_boost_values(region, pool, zk) == (end_time, factor, 0)
```

Comment:
This looks good, but there is quite a bit of room here for type errors. Are you worried about type errors?
Also can you do a namedTuple to give semantic meaning to the "0" at the end there?

Reply:
I'm confident this should work:
>>> (0.000,1.5,2) == (0,1.5,2.00000)
True